src/test/java/org/agbrothers/englishbot/service/LessonServiceTest.java

```diff
@@ -10,96 +12,98 @@
-        lessonRegistry.put(chatId, lesson);
-        lessonService.setLessonRegistry(lessonRegistry);
 
-        Lesson result = lessonService.getLesson(chatId);
+        when(lessonRepository.findLessonByUser(user)).thenReturn(lesson);
+
+        lessonService.getLesson(user);
 
-        assertEquals(lesson, result);
         verify(wordPoolService, never()).getRandomWordPool();
+        verify(lessonRepository, never()).saveAndFlush(any(Lesson.class));
     }
 
     @Test
     void getLesson_NoLesson_ExpectNewLesson() {
-        String chatId = "testChatId";
-        Map<String, Lesson> lessonRegistry = new HashMap<>();
-        lessonService.setLessonRegistry(lessonRegistry);
-        List<Word> words = Collections.singletonList(new Word());
-
-        when(wordPoolService.getRandomWordPool()).thenReturn(words);
+        User user = new User();
+        when(lessonRepository.findLessonByUser(user)).thenReturn(null);
 
-        Lesson result = lessonService.getLesson(chatId);
+        lessonService.getLesson(user);
 
-        assertNotNull(result);
-        assertEquals(result, lessonRegistry.get(chatId));
+        verify(wordPoolService).getRandomWordPool();
+        verify(lessonRepository).saveAndFlush(any(Lesson.class));
     }
 
     @Test
     void getLesson_NoCurrentWordInLesson_ExpectNewLesson() {
         List<Word> words = Collections.singletonList(new Word());
-        Lesson lesson = new Lesson(words, words);
-        String chatId = "testChatId";
-        Map<String, Lesson> lessonRegistry = new HashMap<>();
-        lessonRegistry.put(chatId, lesson);
-        lessonService.setLessonRegistry(lessonRegistry);
+        User user = new User();
+        Lesson lesson = new Lesson(words, words, user);
 
+        when(lessonRepository.findLessonByUser(user)).thenReturn(lesson);
         when(wordPoolService.getRandomWordPool()).thenReturn(words);
 
-        Lesson result = lessonService.getLesson(chatId);
+        lessonService.getLesson(user);
 
-        assertNotNull(result);
-        assertNotEquals(lesson, result);
-        assertEquals(lessonRegistry.get(chatId), result);
-        assertNotEquals(lessonRegistry.get(chatId), lesson);
         verify(wordPoolService).getRandomWordPool();
+        verify(lessonRepository).delete(any(Lesson.class));
+        verify(lessonRepository).saveAndFlush(any(Lesson.class));
     }
 
     @Test
     void createLesson() {
         List<Word> words = Collections.singletonList(new Word());
+        User user = new User();
 
         when(wordPoolService.getRandomWordPool()).thenReturn(words);
 
-        Lesson lesson = lessonService.createLesson();
+        lessonService.createLesson(user);
 
         verify(wordPoolService).getRandomWordPool();
-        assertEquals(lesson.getAnswersPool(), words);
-        assertEquals(lesson.getWordPool(), words);
+        verify(lessonRepository).saveAndFlush(any(Lesson.class));
+    }
+
+    @Test
+    void removeLesson_NoUser_ExpectRemoveUser() {
```

Comment:
the name of the test is not understandable

Reply:
Thanks. Done